addons/mail/static/src/js/activity.js

```diff
@@ -225,6 +250,28 @@ var BasicActivity = AbstractField.extend({
     // Handlers
     //------------------------------------------------------------
 
+    /**
+     * @private
+     */
+    _bindOnUploadAction: function (activities) {
+        var self = this;
+        _.each(activities, function(activity) {
+            if (activity._fileuploadID) {
+                $(window).on(activity._fileuploadID, function() {
```

Comment:
would use handler function, so that you don't need `var self = this`:
```js
$(window).on(activity._fileuploadID, this._onFileUploaded.bind(this));
```

Reply:
But in the potential `_onFileUplaoded` function, I need the activity to it as done. Using the handler breaks the link.